### ISSUE TITLE
[Bug] SimMIM poor result

### ISSUE THREAD

**Issue by okotaku** [open-mmlab/mmselfsup#608]
### Branch

1.x branch (1.x version, such as `v1.0.0rc2`, or `dev-1.x` branch)

### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmselfsup/issues) and [Discussions](https://github.com/open-mmlab/mmselfsup/discussions) but cannot get the expected help.
- [X] I have read the [documentation](https://mmselfsup.readthedocs.io/en/latest/) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version](https://github.com/open-mmlab/mmselfsup).

### Environment

mmcls==1.0.0rc3

### Describe the bug

```
mim test mmcls configs/swin/swin-base_16xb64_in1k.py --checkpoint https://download.openmmlab.com/mmselfsup/1.x/simmim/simmim_swin-base_16xb128-amp-coslr-800e_in1k-192/swin-base_ft-8xb256-coslr-100e_in1k-224/swin-base_ft-8xb256-coslr-100e_in1k-224_20220916-b202cd1c.pth

>> 12/06 06:43:19 - mmengine - [4m[37mINFO[0m - Epoch(test) [782/782]  accuracy/top1: 52.7320  accuracy/top5: 74.4920
```

As I understand it, simim's swin-b finetune and mmcls' swin-b config are the same.
Therefore, this command should give me the proper accuracy.
But in fact the score was very low. I also tried finetune with CUB, but the accuracy was low.

**Comment by YuanLiuuuuuu** [comment]
Thank you for reporting this bug, we will fix it as soon as possible

**Comment by YuanLiuuuuuu** [comment]
This bug is fixed in this [PR](https://github.com/open-mmlab/mmselfsup/pull/622). Besides, we find the link of the ckpt for Swin-L fine-tuning is also problematic. We will fix it as soon as possible.

**Comment by okotaku** [comment]
Thank you!

**Comment by okotaku** [comment]
@YuanLiuuuuuu I think there is also bug in pretrain checkpoint.
Please check it.
My CUB experiments got poor results.

|                Model                 | Top-1 (%) |
| :----------------------------------: | :-------: |
|         swin-b-224-in1k_2xb8_cub          |   90.59   |
|   swin-b-224-simmim-100e_2xb8_cub    |   52.14   |
|   swin-b-224-simmim-800e_2xb8_cub    |   62.58   |
| swin-b-224-simmim-800e-ft1k_2xb8_cub |   77.86   |

**Comment by okotaku** [comment]
And simsiam weights may have same bug.
<img width="438" alt="スクリーンショット 2022-12-08 17 57 32" src="https://user-images.githubusercontent.com/24734142/206403035-22cabee8-9152-4a3f-b18e-7f0a052bfbae.png">

**Comment by YuanLiuuuuuu** [comment]
The recent updated ckpt is obtained by fine-tuning the pretrained ckpt, and we get expected acc on ImageNet-1K. The low acc reported by you, I assume, may be caused by the following two reasons: 1) your fine-tuning recipes are different from [that](https://github.com/open-mmlab/mmselfsup/blob/master/configs/benchmarks/classification/imagenet/swin-base_ft-8xb256-coslr-100e_in1k-224.py) we use, 2) the CUB dataset is an OOD dataset for ImageNet, and such degraded performance can also be observed on some other OOD datasets.

**Comment by okotaku** [comment]
I see.
I don't understand why only simsiam and simmim have low accuracy while mocov3, byol and others record high accuracy.
All resnet50 recipes are the same, only checkpoint is different.
The simmim one is the same as the mmcls CUB recipe, differing only in arch(l -> b) and checkpoint.
https://github.com/open-mmlab/mmclassification/blob/dev-1.x/configs/swin_transformer/swin-large_8xb8_cub_384px.py

For simmim, I will convert the official weights and investigate investigate if the mmcls weight is the bug.

**Comment by okotaku** [comment]
The new weight works well. Thank you!
```
mim test mmcls configs/swin_transformer/swin-base_16xb64_in1k.py --checkpoint https://download.openmmlab.com/mmselfsup/1.x/simmim/simmim_swin-base_16xb128-amp-coslr-800e_in1k-192/swin-base_ft-8xb256-coslr-100e_in1k-224/swin-base_ft-8xb256-coslr-100e_in1k-224_20221208-155cc6e6.pth


> 12/08 09:43:25 - mmengine - INFO - Epoch(test) [782/782]  accuracy/top1: 83.6220  accuracy/top5: 96.5440
```

**Comment by YuanLiuuuuuu** [comment]
Close with being solved. If you have any other problems, feel free to reopen it.
